src/main/java/utils/EventReporter.java

```diff
@@ -0,0 +1,144 @@
+package utils;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.events.WebDriverEventListener;
+
+public class EventReporter implements WebDriverEventListener {
```

Comment:
I think is missing the exercise of the chapter 13 (Add and try with more methods, apart from the beforeClickOn)